### PR TITLE
Branding parameter to prevent duplicated folder sync pair

### PR DESCRIPTION
**Pull request by TheOneRing**
Default:
image
With branding:
image

Fixes: #9523

### CODE REVIEW

File: changelog/unreleased/9523 [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Enhancement: Added brandin parameter to disallow duplicated folder sync pairs
```
Comment (fmoc) [inline changelog/unreleased/9523:1]:
```suggestion
Enhancement: Added branding parameter to disallow duplicated folder sync pairs
```

File: src/gui/folderwizard.cpp [hunk 2]
```diff
@@ -43,17 +43,17 @@
 
 namespace OCC {
 
-QString FormatWarningsWizardPage::formatWarnings(const QStringList &warnings) const
+QString FormatWarningsWizardPage::formatWarnings(const QStringList &warnings, bool error) const
```
Comment (fmoc) [inline src/gui/folderwizard.cpp:46]: I'd call the variable `isError` or `hasError`. We use `error` typically for variables of some Qt error type.

File: src/gui/folderwizard.cpp [hunk 3]
```diff
@@ -43,17 +43,17 @@
...
 {
     QString ret;
     if (warnings.count() == 1) {
-        ret = tr("<b>Warning:</b> %1").arg(warnings.first());
+        ret = error ? tr("<b>Error:</b> %1").arg(warnings.first()) : tr("<b>Warning:</b> %1").arg(warnings.first());
```
Comment (fmoc) [inline src/gui/folderwizard.cpp:50]: Not too readable. Maybe use `if`.

File: src/gui/folderwizard.cpp [hunk 4]
```diff
@@ -43,17 +43,17 @@
...
     } else if (warnings.count() > 1) {
-        ret = tr("<b>Warning:</b>") + " <ul>";
+        QStringList w2;
         for (const auto &warning : warnings) {
-            ret += QString::fromLatin1("<li>%1</li>").arg(warning);
+            w2.append(QStringLiteral("<li>%1</li>").arg(warning));
         }
-        ret += "</ul>";
+        ret = error ? tr("<b>Error:</b><ul>%1</ul>").arg(w2.join(QString())) : tr("<b>Warning:</b><ul>%1</ul>").arg(w2.join(QString()));
```
Comment (fmoc) [inline src/gui/folderwizard.cpp:56]: Seems redundant to the block above.

Comment (TheOneRing) [reply src/gui/folderwizard.cpp:56]: In theory yes but I'm not an expert on rtl languages and whether there might be a case where the position of the argument needs to be changed.
Also this gives the translator more contexts.

**Review by fmoc — APPROVED** [review]

**Comment by ownclouders** [comment]
Results for <strong>GUI-tests</strong> https://drone.owncloud.com/owncloud/client/11106/5/1
GUI Logs: (https://cache.owncloud.com/public/owncloud/client/11106/guiReportUpload/index.html)
Server Logs: (https://cache.owncloud.com/public/owncloud/client/11106/guiReportUpload/serverlog.log)

**Pull request merged**